flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/enumerator/PostgresSourceEnumerator.java

```diff
@@ -56,6 +60,20 @@ public void start() {
      * reading the globalStreamSplit to catch all data changes.
      */
     private void createSlotForGlobalStreamSplit() {
+        SlotState slotInfo = null;
+        try (PostgresConnection connection = postgresDialect.openJdbcConnection()) {
+            slotInfo =
+                    connection.getReplicationSlotState(
+                            postgresDialect.getSlotName(), postgresDialect.getPluginName());
+        } catch (SQLException e) {
+            // do nothing
```

Comment:
It's better to throw a runtime exception, otherwise we may create a duplicated slot  in following code